### PR TITLE
Add an exported serve dashboard json

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
Screenshot 2023-03-16 at 7 57 01 PM
Screenshot 2023-03-16 at 7 57 05 PM

Refactor grafana_dashboard_factory to support multiple dashboards
Update serve http latency metric to use route instead route_prefix as tag

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@alanwguo let's make the QPS graphs consistent. Right now there is total QPS then broken down by route but for errors it's broken down by deployment. I would suggest:

- For both, no need to have a separate plot for total vs. broken down, it can just be one plot w/ multiple lines including a total line (I think you can do that).
- Let's add by route and by deployment for both.

So I guess this means we should have four charts: QPS by route, QPS by deployment, error QPS by route, error QPS by deployment.

WDYT?

**Comment by alanwguo** [comment]
> * For both, no need to have a separate plot for total vs. broken down, it can just be one plot w/ multiple lines including a total line (I think you can do that).
> * Let's add by route and by deployment for both.

I incorporated your feedback with the following changes:
1. got rid of the total qps and total latency charts and moved them to the "By route" charts.
2. Made the HTTP metrics and Deployment metrics more or less mirror each other.

Row 1 + 2: High level cluster utilization metric, request and error QPS and latencies by route
Row 3 + 4: High level deployments metric, request and error QPS and latencies by deployment
Row 5: Various useful hardware 


Screenshot 2023-03-17 at 4 08 10 PM
Screenshot 2023-03-17 at 4 08 13 PM

**Review by edoakes — APPROVED** [review]
Looks awesome, thank you

**Comment by scottsun94** [comment]
The "queue size" in row 5 is also per deployment? Not sure if we make the title more explicit/consistent with others.

If we added those graphs into ray dashboard, it would be inside the application detail page?

File: dashboard/modules/metrics/dashboards/default_dashboard_panels.py [hunk 1]
```diff
@@ -0,0 +1,366 @@
+# flake8: noqa E501
```
Comment (rkooo567) [inline dashboard/modules/metrics/dashboards/default_dashboard_panels.py:1]: Is this file exactly the same as before?

**Comment by alanwguo** [comment]
> I have two questions:
> 
> 1. The "queue size" in row 5 is also per deployment? Not sure if we make the title more explicit/consistent with others.
> 2. If we added those graphs into ray dashboard, it would be inside the application detail page?

1. Yes it is per deployment. I'll update the title

2. Right now we don't have metrics which are tagged by application. We won't be able to support application detail metrics until ray 2.5 I think. For now, we can stick with global level and deployment level.

**Comment by scottsun94** [comment]
> > I have two questions:
> > 
> > 1. The "queue size" in row 5 is also per deployment? Not sure if we make the title more explicit/consistent with others.
> > 2. If we added those graphs into ray dashboard, it would be inside the application detail page?
> 
> 1. Yes it is per deployment. I'll update the title
> 2. Right now we don't have metrics which are tagged by application. We won't be able to support application detail metrics until ray 2.5 I think. For now, we can stick with global level and deployment level.

SG. Thanks!

**Review by ericl — APPROVED** [review]
High level structure lgtm.

**Comment by akshay-anyscale** [comment]
Are the queue size metrics only at the deployment level or do they include queues in HTTPProxy as well?

**Comment by alanwguo** [comment]
> Are the queue size metrics only at the deployment level or do they include queues in HTTPProxy as well?

Only per deployment right now

**Review by sihanwang41 — APPROVED** [review]

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**